### ISSUE TITLE
Avatar Performance Is Decreased

### ISSUE THREAD

**Issue by RyuviTheViali** [Yellow-Dog-Man/Resonite-Issues#2895]
### Describe the bug?

As of version Beta 2024.9.6.626 I've noticed that I've been getting lesser performance than usual, namely with my avatars. So far I've seen this drop in performance across all of my avatars, in any world, regardless of complexity, ProtoFlux present, etc. I use FPS VR to see my render times, and I've noticed on my GPU its usage is higher than previously on average, and I'm seeing an strange repetitive spiking in my render time as well. This spiking is present for me even in my local home with the default avatar applies, albeit a good deal less pronounced.

Overall I'd say my average FPS has dipped by about 15-20 FPS.

### To Reproduce

1. Load Resonite (Desktop or VR)
2. Apply an avatar
3. Notice diminished performance

### Expected behavior

Avatars aren't as GPU performance-heavy.

### Screenshots

This is a screenshot of my FPS VR overlay seen through the Steam VR display showing my GPU frame time being abnormally high in my cloud home.
![2024-09-06_14-25-57](https://github.com/user-attachments/assets/c2d77210-cb5e-4f78-b94a-9251b9401d27)

### Resonite Version Number

Beta 2024.9.6.626

### What Platforms does this occur on?

Windows

### What headset if any do you use?

Vive Pro Eye

### Log Files

[RYUVI-TOWER - 2024.9.6.626 - 2024-09-06 14_55_01.log](https://github.com/user-attachments/files/16912451/RYUVI-TOWER.-.2024.9.6.626.-.2024-09-06.14_55_01.log)

### Additional Context

So far I'm unsure what would be causing this drop in performance, but it certainly wasn't present just last night in the previous version.

### Reporters

Ryuvi (Discord: `ryuvi`)

**Comment by shiftyscales** [comment]
Seeking input from @Frooxius - this seems like something we may have to profile- I also first noticed this on `2024.9.6.626` when I was using it earlier this morning, and did not notice it prior. Trying again now on `2024.9.6.1185` and it's still occurring.

It's a noticeable frame hitch about every four seconds on the dot.

Didn't seem to correspond to any background task

While testing it again just now it also occurs while in the local home in the default head and hands avatar, so I do not suspect it's a matter of avatar / world complexity or any running background processes.

Just now I've recorded the following 120Hz video clips in `Beta 2024.9.6.1189` both in VR (Oculus runtime / Quest), and desktop-only modes, as separate runs.

Observed in VR while looking at the debug dialog- on the dot every ~4 seconds there is a frame spike.

https://github.com/user-attachments/assets/f19cbb0a-aab9-4617-baaa-12d0abb6b467

The same observed in desktop-only mode with MSI Afterburner / RTSS overlays enabled to visualize the frametimes- note the periodic spike consistently every ~4 seconds which also seems to cause the avatar's hand to jump forward briefly.


https://github.com/user-attachments/assets/e9ba4766-7646-4af3-a1b5-78d62bebd336

**Comment by ohzee00** [comment]
Shot in the dark but could the render transform overrides that take every skin mesh renderer for the new locomotion system be a culprit? I know you tested in local and such but I believe the default avatar has it applied on it too

**Comment by shiftyscales** [comment]
I don't believe so @ohzee00, those were in for much earlier than two builds ago I think it would've been likely to have gotten caught at some point prior.

**Comment by RyuviTheViali** [comment]
For me, the regular raise in frame time is about 5-7 times a second, constantly - I haven't noticed the four-second spikes just yet.

**Comment by Frooxius** [comment]
Can you check if 2024.9.10.682 makes any difference?

**Comment by shiftyscales** [comment]
Things seem to be considerably improved and quite smooth for me @Frooxius. Pending @RyuviTheViali's confirmation I think this can be closed.

**Comment by Zyro1331** [comment]
I have generally noticed a slight raise in performance, but I have yet to be in VR to truly see my framerate difference. 

I have also learned that small lag spike every 4 seconds appears is actually the world browser refreshing in the background.
It does this even while the dashboard is closed or the tab is closed. 

(In-game framerate counter showing the lagspikes)
![image](https://github.com/user-attachments/assets/5b745833-f4e4-407d-a7e1-fc108e11a7f3)

**Comment by ohzee00** [comment]
> @shiftyscales I have also learned that small lag spike that occurs every 4 seconds is actually the world browser refreshing in the background. It does this even while the dashboard is closed or the tab is closed.
> 
> (In-game framerate counter showing the lag-spikes) ![image](https://private-user-images.githubusercontent.com/49619936/366257308-5b745833-f4e4-407d-a7e1-fc108e11a7f3.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjYwMTU5MTgsIm5iZiI6MTcyNjAxNTYxOCwicGF0aCI6Ii80OTYxOTkzNi8zNjYyNTczMDgtNWI3NDU4MzMtZjRlNC00MDdkLWE3ZTEtZmMxMDhlMTFhN2YzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTExVDAwNDY1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWFmMjI2MzAxOWY0ZDU2MjExZGUxMWY2MmIyNTkxOWY0YjQ2MDI4NmYwMGYzM2RkZmYzNGNiMTBhMjJkZDhmYzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WT70QxD2AE_EQmWT8U9SzHTJ4fujKuidWWDqbyx_Sks)

Can you test that theory by removing the world browser facet and seeing if that spike is visible still?

**Comment by RyuviTheViali** [comment]
After testing in Desktop and VR, I can say my performance is a good deal better than it was, so I feel this issue has been resolved.

**Comment by Zyro1331** [comment]
> @ohzee00 Can you test that theory by removing the world browser facet and seeing if that spike is visible still?

Here's a screen capture showing my in-game FPS counter, as you can see when the refresh animation plays, the game has a small dip in frametime. 

https://github.com/user-attachments/assets/c72a95d0-b297-4f40-8bf3-c96d6098780f

Now here's a capture after removing all the facets on the Worlds tab entirely to prevent it from refreshing. The 4-second lag spikes are completely gone. 

https://github.com/user-attachments/assets/7689f9e9-a64e-4ea9-b4c6-0512a33eeaee

**Comment by shiftyscales** [comment]
That doesn't seem like it's related to this issue, @Zyro1331. As indicated, performance is back to where it was for me. When the dash is closed / not focused on the Worlds tab it should be inactive. For me it'd be expected there'd be some amount of hit while refreshing as it'd need to remove/regenerate the new sessions.

If you feel it is anomalous behaviour though, I'd recommend opening a new issue to track it.
